GlobalOptions.py

```diff
@@ -1,54 +1,46 @@
 from abc import ABC, abstractmethod
 
-
-class GlobalOptionBuilder(ABC):
-    globalOptionsList = []
-
+class GlobalOptionsBuilder(ABC):
+    '''Abstract Component Class'''
     @abstractmethod
-    def add_option(self, **kwargs):
+    def add_options(self):
         pass
 
+class DefaultGlobalOptions(GlobalOptionsBuilder):
```

Comment:
You can initialize a blank list here, if you pass it when creating an object, people won't understand what is going on !
 ```
self.globalOptionsList = []
```